src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1623,6 +1652,21 @@ internal static SpecialType GetEnumPromotedType(SpecialType underlyingType)
                 return ConstantValue.Create(newValue, resultType);
             }
 
+            try
+            {
+                newValue = FoldNativeIntegerBinaryOperator(kind, valueLeft, valueRight);
+            }
+            catch (OverflowException)
+            {
+                Error(diagnostics, ErrorCode.ERR_CheckedOverflow, syntax);
```

Comment:
>Error(diagnostics, ErrorCode.ERR_CheckedOverflow, syntax); [](start = 16, length = 58)

I would expect this the error reported conditionally. Only when CheckOverflowAtCompileTime is true. An error in unchecked context is unexpected, we simply should evaluate the operation at runtime.